src/simulations/PowerPerturbation.jl

```diff
@@ -21,7 +21,7 @@ function (pd::PowerPerturbation)(powergrid)
     #TODO: check if node type supported for power drop
     node_list_power_drop = copy(powergrid.nodes)
     node_for_drop = node_list_power_drop[pd.node_number]
-    node_for_drop = @set node_for_drop.P *= pd.fraction
+    node_for_drop = @set node_for_drop.P = round(Int64, node_for_drop.P*pd.fraction)
```

Comment:
Why do you round here? Presumably we want to set the (now floating point) `de_for_drop.P` to `node_for_drop.P*pd.fraction`?

Reply:
Thinking a bit further about this, i don't think that it is a good solution. In the test we set the nodes P to an `Int`. That is why we get a type instability after applying the drop, cause it is a `Float` after multiplication with the fraction. This fixes the test but doesn't really solve the problem. Additionally we loose precision because of rounding.
A real solution would be to set parameter `P`'s type to `Float` in the node definition and disallow dynamic typing here.
Or do you have another idea @FHell ?

Reply:
There are two changes in the commit, one changes the parameter P to Float in line 5, the other introduces the rounding. Did you try with just the former, without the rounding and cast to Int? I might get some time to play around with this next week.

I also agree with fixing the types of the parameters. The only scenario where we might need type flexibility is with ForwardDifferentiation with respect to parameters. Not an immediate concern.

Reply:
Of course it would work just to set the P parameters as float in the test, but that would be a brittle solution.
Other users might easily put `Int` values there as well and run into the same error.

Reply:
I think as written now PowerPerturbation doesn't do what we want it to do. If the original power is set to 0.7 and we do a 20% drop we get a power of: `round(Int64, 0.7 * 0.8) == 1`. Further I don't fully understand how this is type stable, as it seems like it replaces a Float `SwingEqLVS(H=1., P=-1.0 ...)` with an Int (before the code was replacing an Int with a Float, so probably Int to Float is cast automatically...).

I would support the solution of enforcing Float for all parameters.

The only other robust solution I can see would be to make the parameter structs mutable, so we can reach in and change them without swapping out f. This way the error would be clearer, one would produce an inexact error exception when trying to set `...nodes[i].f.P *= pd.fraction`. We could catch that and tell people why things failed.

Reply:
Another solution:
```julia
newP = typeof(node_for_drop.P)(node_for_drop.P * pd.fraction)
node_for_drop = @set node_for_drop.P = newP
```

This will error this way:

```julia
typeof(1)(1 * 0.7)
ERROR: InexactError: Int64(0.7)
```